app/controllers/spree/paypal_controller_decorator.rb

```diff
@@ -51,6 +56,32 @@ def express
     end
   end
 
+  def confirm
+    @order = current_order || raise(ActiveRecord::RecordNotFound)
+
+    # At this point the user has come back from the Paypal form, and we get one
+    # last chance to interact with the payment process before the money moves...
+    return handle_insufficient_stock unless sufficient_stock?
+
+    @order.payments.create!({
+      source: Spree::PaypalExpressCheckout.create({
+        token: params[:token],
+        payer_id: params[:PayerID]
+      }),
+      amount: @order.total,
+      payment_method: payment_method
+    })
+    @order.next
+    if @order.complete?
+      flash.notice = Spree.t(:order_processed_successfully)
+      flash[:commerce_tracking] = "nothing special"
```

Comment:
Is this flash not user-visible?

Reply:
yes, and what's even that `:commerce_tracking`?

Reply:
This is the code directly imported from the gem :see_no_evil:  I've made this issue to pull it all in and clean it up: #6421

Reply:
I thought so but I looked it up in better_spree_paypal_express and couldn't find any result: https://github.com/openfoodfoundation/better_spree_paypal_express/search?q=commerce_tracking. Am I missing something?

Reply:
Couldn't find it searching either but found it browsing the code here https://github.com/openfoodfoundation/better_spree_paypal_express/blob/2-1-0-stable/app/controllers/spree/paypal_controller.rb#L62